src/Type/Symfony/Config/NodeDefinitionEndDynamicReturnTypeExtension.php

```diff
@@ -0,0 +1,39 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type\Symfony\Config;
+
+use PhpParser\Node\Expr\MethodCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\MethodReflection;
+use PHPStan\Reflection\ParametersAcceptorSelector;
+use PHPStan\Type\DynamicMethodReturnTypeExtension;
+use PHPStan\Type\Type;
+
+final class NodeDefinitionEndDynamicReturnTypeExtension implements DynamicMethodReturnTypeExtension
```

Comment:
This class looks exactly the same as NodeBuilderEndDynamicReturnTypeExtension . Instead of duplicating the whole class, you can have only a single class, but making it dynamic by asking for $class in the constructor, then registering the class as two services in extension.neon, with different arguments.

Reply:
Much better indeed.